testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+        for (String ff : allFieldName) {
+            writer.println(ff);
+        }
+        writer.close();
+
+        for (Class c : loadedClasses) {
+            // Ignore classes in standard java to get top-level
+            // TODO(gyori): make this read from file or config option
+            String clz = c.getName();
+            if ((clz.contains("java.") && !clz.startsWith("java.lang.System"))
+                || (clz.contains("javax.") && !clz.startsWith("javax.cache.Caching"))
+                || clz.contains("javafx.")
+                || clz.contains("jdk.")
+                || clz.contains("scala.")
+                || clz.contains("sun.")
+                || clz.contains("edu.illinois.cs")
+                || clz.contains("org.custommonkey.xmlunit")
+                || clz.contains("org.junit")
+                || clz.contains("statecapture.com.")
+                || clz.contains("statecapture.org.")
+                || clz.equals("com.openpojo.reflection.impl.AClassWithBadMethod__Generated_OpenPojo")) {
+                continue;
+            }
+
+            Set<Field> allFields = new HashSet<Field>();
+            try {
+                Field[] declaredFields = c.getDeclaredFields();
+                Field[] fields = c.getFields();
+                allFields.addAll(Arrays.asList(declaredFields));
+                allFields.addAll(Arrays.asList(fields));
+            } catch (NoClassDefFoundError e) {
+                continue;
+            } catch (Exception exception) {
+                continue;
+            }
+            // prepare for the subxml fold
+
+            for (Field f : allFields) {
+                String fieldName = getFieldFQN(f);
+
+                String subFieldName = fieldName.substring(fieldName.lastIndexOf(".")+1, fieldName.length());
+                if (ignores.contains(fieldName)) {
+                    continue;
+                }
+
+                // if a field is final and has a primitive type there's no point to capture it.
+                if (Modifier.isStatic(f.getModifiers())
+                    && !(Modifier.isFinal(f.getModifiers()) &&  f.getType().isPrimitive())) {
+                    try {
+                        if (shouldCapture(f)) {
+                            f.setAccessible(true);
+
+                            Object instance;
+                            try {
+                                instance = f.get(null);
+                            } catch (NoClassDefFoundError NCDFE) {
+                                instance = null;
+                                NCDFE.printStackTrace();
+                            }
+                            if(instance instanceof ThreadLocal) {
+                                instance = ((ThreadLocal)instance).get();
+                            }
+                            LinkedHashMap<String, Object> nameToInstance_temp = new LinkedHashMap<String, Object>();
```

Comment:
Maybe we don't need to make some temp map to serialize first to check dirty or not, we could just directly try to serialize the object and if that is dirty we ignore it.